src/backend/src/services/projects.services.ts

```diff
@@ -0,0 +1,394 @@
+      userId,
+      wbsElementId
+    );
+    const projectLeadChangeJson = createChangeJsonNonList(
+      'project lead',
+      await getUserFullName(originalProject.wbsElement.projectLeadId),
+      await getUserFullName(projectLead),
+      crId,
+      userId,
+      wbsElementId
+    );
+
+    // add to changes if not undefined
+    if (nameChangeJson !== undefined) {
+      changes.push(nameChangeJson);
+    }
+    if (budgetChangeJson !== undefined) {
+      changes.push(budgetChangeJson);
+    }
+    if (summaryChangeJson !== undefined) {
+      changes.push(summaryChangeJson);
+    }
+    if (driveChangeJson !== undefined) {
+      changes.push(driveChangeJson);
+    }
+    if (slideChangeJson !== undefined) {
+      changes.push(slideChangeJson);
+    }
+    if (bomChangeJson !== undefined) {
+      changes.push(bomChangeJson);
+    }
+    if (taskChangeJson !== undefined) {
+      changes.push(taskChangeJson);
+    }
+    if (projectManagerChangeJson !== undefined) {
+      changes.push(projectManagerChangeJson);
+    }
+    if (projectLeadChangeJson !== undefined) {
+      changes.push(projectLeadChangeJson);
+    }
+
+    // Dealing with lists
+    const rulesChangeJson = createRulesChangesJson('rules', originalProject.rules, rules, crId, userId, wbsElementId);
+    const goalsChangeJson = createDescriptionBulletChangesJson(
+      originalProject.goals.filter((element) => !element.dateDeleted).map((element) => descBulletConverter(element)),
+      goals,
+      crId,
+      userId,
+      wbsElementId,
+      'goals'
+    );
+    const featuresChangeJson = createDescriptionBulletChangesJson(
+      originalProject.features.filter((element) => !element.dateDeleted).map((element) => descBulletConverter(element)),
+      features,
+      crId,
+      userId,
+      wbsElementId,
+      'features'
+    );
+    const otherConstraintsChangeJson = createDescriptionBulletChangesJson(
+      originalProject.otherConstraints
+        .filter((element) => !element.dateDeleted)
+        .map((element) => descBulletConverter(element)),
+      otherConstraints,
+      crId,
+      userId,
+      wbsElementId,
+      'other constraints'
+    );
+    // add the changes for each of dependencies, expected activities, and deliverables
+    changes = changes
+      .concat(rulesChangeJson)
+      .concat(goalsChangeJson.changes)
+      .concat(featuresChangeJson.changes)
+      .concat(otherConstraintsChangeJson.changes);
+
+    // update the project with the input fields
+    const updatedProject = await prisma.project.update({
+      where: {
+        wbsElementId
+      },
+      data: {
+        budget,
+        summary,
+        googleDriveFolderLink,
+        slideDeckLink,
+        bomLink,
+        taskListLink,
+        rules,
+        wbsElement: {
+          update: {
+            name,
+            projectLeadId: projectLead,
+            projectManagerId: projectManager
+          }
+        }
+      },
+      ...projectQueryArgs
```

Comment:
I usually replaced ...manyRelationArgs with ...projectQueryArgs, but I don't think there was a ...manyRelationArgs here before, is there a reason why we need to add in the ...projectQueryArgs? Or is that something we should always do